packages/multichain-account-service/src/MultichainAccountWallet.ts

```diff
@@ -173,4 +183,116 @@ export class MultichainAccountWallet<
   getMultichainAccountGroups(): MultichainAccountGroup<Account>[] {
     return Array.from(this.#accounts.values()); // TODO: Prevent copy here.
   }
+
+  /**
+   * Gets next group index for this wallet.
+   *
+   * @returns The next group index of this wallet.
+   */
+  getNextGroupIndex(): number {
+    // We do not check for gaps.
+    return (
+      Math.max(
+        -1, // So it will default to 0 if no groups.
+        ...this.#accounts.keys(),
+      ) + 1
+    );
+  }
+
+  /**
+   * Creates a multichain account group for a given group index.
+   *
+   * @param groupIndex - The group index to use.
+   * @throws If any of the account providers fails to create their accounts.
+   * @returns The multichain account group for this group index.
+   */
+  async createMultichainAccountGroup(
+    groupIndex: number,
+  ): Promise<MultichainAccountGroup<Account>> {
+    const nextGroupIndex = this.getNextGroupIndex();
+    if (groupIndex > nextGroupIndex) {
+      throw new Error(
+        `You cannot use a group index that is higher than the next available one: expected <=${nextGroupIndex}, got ${groupIndex}`,
+      );
+    }
+
+    let group = this.getMultichainAccountGroup(groupIndex);
+    if (group) {
+      // If the group already exists, we just `sync` it and returns the same
+      // reference.
+      group.sync();
+
+      return group;
+    }
+
+    const results = await Promise.allSettled(
+      this.#providers.map((provider) =>
+        provider.createAccounts({
+          entropySource: this.#entropySource,
+          groupIndex,
+        }),
+      ),
+    );
+
+    // READ THIS CAREFULLY:
+    // Since we're not "fully supporting multichain" for now, we still rely on single
+    // :accountCreated events to sync multichain account groups and wallets. Which means
+    // that even if of the provider fails, some accouns (on the `AccountsController`
+    // might be showing up AND that some multichain account wallets/groups might have
+    // been automatically sync'd too.
+    //
+    // We will need to unsubscribe from :accountAdded once we "fully support multichain"
+    // (meaning, the entire codebase relies on this new service for multichain account
+    // creations).
+    // --------------------------------------------------------------------------------
+
+    // If any of the provider failed to create their accounts, then we consider the
+    // multichain account group to have failed too.
+    // NOTE: Though, we don't rollback existing created accounts and that's totally fine
+    // because account creations is assumed to be imdepotent, thus, trying again should
+    // just create the missing accounts, and previously created accounts will just be
+    // re-used as is.
```

Comment:
Do you mind re-writing this comment? It's a bit hard to follow (multi-line comment might be better here). Would appreciate if you expand on these two statements as well:

1. "that even if of the provider fails, some accouns (on the `AccountsController` might be showing up AND that some multichain account wallets/groups might have been automatically sync'd too."
2. "NOTE: Though, we don't rollback existing created accounts and that's totally fine because account creations is assumed to be imdepotent, thus, trying again should just create the missing accounts, and previously created accounts will just be re-used as is.

Reply:
I tried something different, LMK what you think: https://github.com/MetaMask/core/pull/6222/commits/e9fbdff5c2ae772299a28aeab360568604d58d40

Reply:
Yes, better -- thanks!